src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -166,108 +117,128 @@ public J visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ct
-                                classlessTypeName,
-                                classlessTypeName
-                        )
-
-                );
-
-                J.VariableDeclarations mockField = (J.VariableDeclarations) classDecl.getBody().getStatements().get(0);
-                mockedTypesIdentifiers.add(mockField.getVariables().get(0).getName());
+                return m;
             }
-            getCursor().putMessage(MOCKED_TYPES_FIELDS, mockedTypesIdentifiers);
 
-            maybeAutoFormat(classDecl, classDecl.withPrefix(classDecl.getPrefix().
-                    withWhitespace("")), classDecl.getName(), ctx, getCursor());
-            maybeAddImport(MOCKED_STATIC);
-            maybeAddImport("org.mockito.Mockito", "mockStatic");
-            return classDecl;
-        }
+            AnnotationMatcher setUpAnnotationMatcher = new AnnotationMatcher(
+              setUpMethodAnnotationSignature);
+            if (m.getAllAnnotations().stream().anyMatch(setUpAnnotationMatcher::matches)) {
+                // Move the mockStatic method to the setUp method
+                Map<String, J.MethodInvocation> mockStaticInvocations = getCursor().getNearestMessage(MOCK_STATIC_INVOCATIONS);
 
-        private static boolean isFieldAlreadyDefined(J.Block classBody, String fieldName) {
-            for (Statement statement : classBody.getStatements()) {
-                if (statement instanceof J.VariableDeclarations) {
-                    for (J.VariableDeclarations.NamedVariable namedVariable : ((J.VariableDeclarations) statement).getVariables()) {
-                        if (namedVariable.getSimpleName().equals(fieldName)) {
-                            return true;
+                if (mockStaticInvocations != null) {
+                    for (Map.Entry<J.Identifier, Expression> mockedTypesFieldEntry : getMockedTypesFields().entrySet()) {
+                        // Only add close method invocation if not already exists
+                        J.Block methodBody = m.getBody();
+                        if (methodBody == null || isStaticMockAlreadyOpened(mockedTypesFieldEntry.getKey(), methodBody)) {
+                            continue;
+                        }
+
+                        String className = mockedTypesFieldEntry.getValue().toString();
+                        J.MethodInvocation methodInvocation = mockStaticInvocations.get(className);
+                        if (methodInvocation != null) {
+                            m = m.withBody(methodBody.withTemplate(
+                              JavaTemplate.builder(() -> getCursor().getParentTreeCursor(),
+                                  "mocked#{any(org.mockito.MockedStatic)} = #{any(org.mockito.Mockito)};")
+                                .javaParser(() -> JavaParser.fromJavaVersion()
+                                  .classpathFromResources(ctx, "mockito-core-3.*")
+                                  .build())
+                                .build(),
```

Comment:
Possibly missing imports for `MockedStatic` and `Mockito`. I'm not 100% sure from code review alone if it makes a difference in this context, but since the tests are showing some missing type attribution, I'll point this out as a possible source.